miner/agent.go

```diff
@@ -43,13 +43,17 @@ func NewCpuAgent(chain consensus.ChainReader, engine consensus.Engine) *CpuAgent
 		chain:  chain,
 		engine: engine,
 		stop:   make(chan struct{}, 1),
-		workCh: make(chan *Work, 1),
+		taskCh: make(chan *Package, 1),
 	}
 	return agent
 }
 
-func (self *CpuAgent) Work() chan<- *Work            { return self.workCh }
-func (self *CpuAgent) SetReturnCh(ch chan<- *Result) { self.returnCh = ch }
+func (self *CpuAgent) AssignTask(p *Package) {
+	if atomic.LoadInt32(&self.started) == 1 {
+		self.taskCh <- p
```

Comment:
This is a bit racey (the original code was racey too). It might happen that a CPUAgent is being shut down during AssignTask, which would cause the channel send to hang.

Although we could in theory fix this in various ways... do we actually need the CpuAgent any more? Since we moved both local and remote mining logic into the consensus engines, I think this entire file could be deleted, simplifying a lot of the miner code.

Perhaps I'm missing something?